makelib/kubebuilder.mk

```diff
@@ -63,4 +101,16 @@ $(KUBEBUILDER):
 	@rm -fr $(TOOLS_HOST_DIR)/tmp
 	@$(OK) installing kubebuilder $(KUBEBUILDER_VERSION)
 
+$(CONTROLLERGEN):
+	@$(INFO) installing controller-gen @$(CONTROLLER_GEN_VERSION)
+	@mkdir -p $(TOOLS_HOST_DIR)/tmp-controllergen || $(FAIL)
+
+	@# `go get` only supports versioned go packages when GO111MODULE=on 
+	@#  since $(TOOLS_HOST_DIR) is under $(GO_PROJECT), make a temp folder which has a go.mod, 
+	@#  so that $(GO_PROJECT)/go.mod doesn't get modified because of running `go get`
+	@cd $(TOOLS_HOST_DIR)/tmp-controllergen; rm -f go.mod; GO111MODULE=on $(GOHOST) mod init tmp-controllergen
+	@cd $(TOOLS_HOST_DIR)/tmp-controllergen; GOPATH=$(abspath $(GO_PKG_DIR)) GO111MODULE=on GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION) || $(FAIL)
+	@rm -fr $(TOOLS_HOST_DIR)/tmp-controllergen
+
+	@$(OK) installing controller-gen @$(CONTROLLER_GEN_VERSION)
```

Comment:
can you not use the same `@GO111MODULE=off` approach as in golang.mk?

Reply:
As I have mentioned in the comments a few line earlier, versioned pkgs are only supported when `GO111MODULE=on`

Reply:
No, as the comment above mentions versioned go packages do need `GO111MODULE=on`.

Reply:
I thought about downloading the binary directly, but there is no binary distribution for this tool: https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.2.1

so we do need to build the go src and install it, by using `go get`